node/components.go

```diff
@@ -72,8 +72,9 @@ func baseComponents(cfg *Config, store Store) fx.Option {
 		// state components
 		fx.Provide(statecomponents.NewService),
 		fx.Provide(statecomponents.CoreAccessor(cfg.Core.GRPCAddr)),
-		// RPCServer component
+		// RPC components
 		fx.Provide(rpc.ServerComponent(cfg.RPC)),
+		fx.Provide(rpc.HandlerComponents),
```

Comment:
nit: if you are going to remove Handler from the Node, I propose to use fx.Invoke here. It only calls the function without Providing anything to DI. Basically, that function would only register handlers on the server, and that's all.